src/query_graph/path_tree.rs

```diff
@@ -111,13 +111,59 @@ impl OpPathTree {
         }
         Ok(true)
     }
+
+    fn fmt_internal(
+        &self,
+        f: &mut Formatter<'_>,
+        indent: &String,
+        include_conditions: bool,
+    ) -> std::fmt::Result {
+        if self.is_leaf() {
+            return writeln!(f, "{}", self.vertex());
+        }
+        writeln!(f, "{}:", self.vertex())?;
+        let child_indent = indent.to_owned() + "  ";
+        for child in self.childs.iter() {
+            let index = match child.edge {
+                Some(index) => index,
+                None => EdgeIndex::end(),
+            };
+            write!(f, "{indent} -> [{}] ", index.index())?;
+            if include_conditions {
+                if let Some(ref child_cond) = child.conditions {
+                    write!(f, "!! {{\n{indent} ")?;
+                    child_cond.fmt_internal(f, &child_indent, /*include_conditions*/ true)?;
+                    write!(f, "\n{indent} }}")?;
+                }
+            }
+            write!(f, "{} = ", child.trigger)?;
+            child
+                .tree
+                .fmt_internal(f, &child_indent, include_conditions)?;
+        }
+        Ok(())
+    }
+}
+
+impl Display for OpPathTree {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let indent = "".to_owned(); // Empty indent at the root level
+        self.fmt_internal(f, &indent, /*include_conditions*/ false)
+    }
 }
 
 impl<TTrigger, TEdge> PathTree<TTrigger, TEdge>
 where
     TTrigger: Eq + Hash,
     TEdge: Copy + Hash + Eq + Into<Option<EdgeIndex>>,
 {
+    /// Returns the `QueryGraphNode` represented by `self.node`.
+    /// - Note: This is named after the JS implementation's `vertex` field.
```

Comment:
We conventionally use `PORT_NOTE` to point out differences between the implementations, I'm not sure which of the two fits the best, but as long as we say `PORT_NOTE` or `TODO` we will be able to easily find places like this to revisit later.
```suggestion
    /// PORT_NOTE: This is named after the JS implementation's `vertex` field.
```